src/test/java/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1803,40 +1771,44 @@ public void throwLimitExceededException() {
+    // However adding involves adding a new substudy.
     @Test
-    public void updateIdentifiersFiltersSubstudyPermissions() {
-        // caller in question is in substudyB
-        BridgeUtils.setRequestContext(
-                new RequestContext.Builder().withCallerSubstudies(ImmutableSet.of("substudyA")).build());
+    public void updateIdentifiersCanAddExternalIdInOtherSubstudy() {
+        // caller in question is in substudyA
+        BridgeUtils.setRequestContext(new RequestContext.Builder()
+                .withCallerSubstudies(ImmutableSet.of(SUBSTUDY_ID)).build());
         
-        // account in question is in substudyA
-        mockHealthCodeAndAccountRetrieval();
-        when(externalIdService.getExternalId(TEST_STUDY, EXTERNAL_ID)).thenReturn(Optional.of(extId));
+        // account is associated to the EXTERNAL_ID in SUBSTUDY_ID
+        mockHealthCodeAndAccountRetrieval(EMAIL, null, EXTERNAL_ID);
+        
+        // extId is not in the same substudy. This should work
+        extId.setSubstudyId("anotherSubstudy");
+        extId.setIdentifier("newExtId");
+        when(externalIdService.getExternalId(TEST_STUDY, "newExtId")).thenReturn(Optional.of(extId));
         
-        account.setAccountSubstudies(ImmutableSet.of(AccountSubstudy.create(STUDY.getIdentifier(), "substudyB", ID)));
         when(accountDao.authenticate(STUDY, EMAIL_PASSWORD_SIGN_IN)).thenReturn(account);
         
-        try {
-            // And so this fails...
-            IdentifierUpdate update = new IdentifierUpdate(EMAIL_PASSWORD_SIGN_IN, null, null, EXTERNAL_ID);
-            participantService.updateIdentifiers(STUDY, CONTEXT, update);
-            fail("Should have thrown exception");
-        } catch(EntityNotFoundException e) {
-            assertEquals(e.getMessage(), "Account not found.");
-        }
-        verify(accountDao, never()).updateAccount(any(), any());
+        IdentifierUpdate update = new IdentifierUpdate(EMAIL_PASSWORD_SIGN_IN, null, null, "newExtId");
+        participantService.updateIdentifiers(STUDY, CONTEXT, update);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture(), any());
+        
+        assertEquals(accountCaptor.getValue().getAccountSubstudies().size(), 2);
+        RequestContext context = BridgeUtils.getRequestContext();
+        assertEquals(context.getCallerSubstudies(), ImmutableSet.of(SUBSTUDY_ID, "anotherSubstudy"));
     }
     
     @Test
     public void updateIdentifiersAssignsExternalIdEvenWhenAlreadyAssigned() {
         // Fully associated external ID can be changed by an update.
+        BridgeUtils.setRequestContext(new RequestContext.Builder()
+                .withCallerSubstudies(ImmutableSet.of("substudyB")).build());
         mockHealthCodeAndAccountRetrieval();
-        AccountSubstudy as = AccountSubstudy.create(STUDY.getIdentifier(), "substudyB", ID);
+        AccountSubstudy as = AccountSubstudy.create(TEST_STUDY_IDENTIFIER, "substudyB", ID);
```

Comment:
Do we also have a test that updateIdentifiers will ignore if we attempt to assign the same externalId to the same account?

Reply:
Added updateIdentifiersDoesNothingWhenExternalIdResubmitted() and renamed updateIdentifiersDoesNotReassignExistingExternalId() to updateIdentifiersDoesNotReassignExternalIdOnOtherUpdate() to slightly clarify the difference between the two tests.